intra_process_demo/README.md

```diff
@@ -0,0 +1,171 @@
+## **What Is This?**
+
+This demo is designed to showcase how developers can **manually compose ROS 2 nodes** by **defining them separately** but **combining them in varied process layouts**.
+
+These can be done while avoiding code overhauls or performance restrictions. 
+
+This package consists of the following ROS 2 nodes:
+
+1. `camera_node`
+2. `watermark_node`
+3. `image_view_node`
+4. `image_pipeline_all_in_one`
+5. `two_node_pipeline`
+6. `cyclic_pipeline`
+7. `image_pipeline_with_two_image_view`
+
+Through the use of **intra-process** (as opposed to **inter-process** node communication), lower latency and thus **higher efficiency** is observed for ROS 2 topologies that utilizes this manner of communication. 
+
+These improvements in latency reduction is even **more pronounced when applied to ROS 2 systems with more complex topology**. 
+
+## **Build**
+
+```bash
+colcon build --packages-up-to intra_process_demo
+```
+
+## **Run**
+
+### 1. Two Node Pipeline
+
+Run `two_node_pipeline` via the commands below:
+
+```bash
+ros2 run intra_process_demo two_node_pipeline
+```
+
+> This sets up two nodes, a ROS 2 node that publishes a string with an incremeting integer value, as well as a ROS 2 node that subscribes to print the received string.
+
+![](img/two_node_pipeline.png)
+
+### 2. Cyclic Pipeline
+
+Run `cyclic_pipeline` via the commands below:
+
+```bash
+ros2 run intra_process_demo cyclic_pipeline
+```
+
+> Similar to the previous, instead of creating a new message for each new iteration, the publisher and subscriber nodes only ever use one message instance. This is achieved by having a cycle in the graph and kickstarting the communication externally by having one of the nodes publish before spinning executor.
```

Comment:
```suggestion
> Similar to the previous, instead of creating a new message for each new iteration, the publisher and subscriber nodes only ever use one message instance. This is achieved by having a cycle in the graph and kickstarting the communication externally by having one of the nodes publish before spinning the executor.
```

Reply:
Done.